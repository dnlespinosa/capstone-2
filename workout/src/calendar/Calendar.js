import React, { useContext, useState, useEffect, useRef } from 'react';
import { format, parse, startOfWeek, endOfWeek, addDays, addMonths, subMonths, startOfMonth, endOfMonth, isSameMonth, isSameDay, isToday } from 'date-fns';
import './Calendar.css';
import { Link } from 'react-router-dom';
import UserContext from '../auth/UserContext';
import WorkoutApi from '../api/api';
import Workout from '../Program/Workout';

const Calendar = () => {
    const [currentMonth, setCurrentMonth] = useState(new Date())
    const [selectedDate, setSelectedDate] = useState(new Date())
    const [isWorkout, setIsWorkout] = useState(false)
    const [workout, setWorkout] = useState([])

    const { currentUser, todaysWorkout } = useContext(UserContext)

    const myRef = useRef(null)

    useEffect(function getWorkoutForPage(){
        async function getWorkoutFor(){
            if (currentUser.workouts) {
                for (let i=0; i<currentUser.workouts.length; i++) {
                    let res = await WorkoutApi.getWorkoutById(currentUser.workouts[i].workout_id)
                    setWorkout( (workoutState) => {
                        return [...workoutState, res]
                    })
                }
             } else {
                console.log('loading?')
             }
        }
        getWorkoutFor()
    }, [])

    console.log(currentUser)

    

    const renderHeader = () => {
        const dateFormat = 'MMMM YYYYY';

        return (
            <div className ='header row flex-middle'>
                <div className='col col-start'>
                    <div className='icon' onClick={prevMonth}>
                        chevron_left
                    </div>
                </div>
                <div className='col col-center'>
                    <span>
                        {format(currentMonth, dateFormat)}
                    </span>
                </div>
                <div className='col col-end' onClick={nextMonth}>
                    <div className='icon'>chevron_right</div>
                </div>
            </div>
        )
    };

    const renderDays = () => {
        const dateFormat = 'dddd';
        const days = [];

        let startDate = startOfWeek(currentMonth);
        for (let i=0; i< 7; i++) {
            days.push(
                <div className='col col-center' key={i}>
                    {format(addDays(startDate, i), dateFormat)}
                </div>
            )
            return <div className='days row'>{days}</div>
        }
    }
    const renderCells = () => {
        const monthStart = startOfMonth(currentMonth)
        const monthEnd = endOfMonth(monthStart)
        const startDate = startOfWeek(monthStart);
        const endDate = endOfWeek(monthEnd);

        const dateFormat = 'd';
        const rows = [];

        let days = [];
        let day = startDate;
        let formattedDate = "";

        
        while (day <= endDate) {
            for (let i=0; i<7; i++){

                let displayWorkout
                if (workout.length > 1) {
                    for (let j=0; j<workout.length; j++) {
                        if (workout[j].day_of_month === i)
                            
                            displayWorkout = workout[j].name
                    }
                } else {
                    if (workout.day_of_month === i) {
                        displayWorkout = workout.name
                    }
                }

                // if (displayWorkout != undefined && isSameDay(day, selectedDate)) {
                //     todaysWorkout(displayWorkout)
                // }


                formattedDate = format(day, dateFormat);
                const cloneDay = day;
                days.push(
                    <div className={`col cell ${!isSameMonth(day, monthStart)
                        ? 'disabled'
                        : isSameDay(day, selectedDate) ? 'selected' : ''}`}
                        key={day}
                        onClick={() => onDateClick(cloneDay)}>
                            <div>
                                <Link to={`/personal-workout/${displayWorkout}`}>
                                    <p id='displayedWorkout'>{displayWorkout}</p>
                                </Link> 
                                
                            </div>
                            <span className='number'>{formattedDate}</span>
                            {/* <span className='bg'>{formattedDate}</span> */}
                        </div>
                );
                day = addDays(day, 1)
            }
            rows.push(
                <div className='row' key={day}>
                    {days}
                </div>
            )
            days = [];
        }
        return <div className='body'>{rows}</div>
    }

    const onDateClick = (day) => {
        setSelectedDate(day)
    }
    const nextMonth = () => {
        setCurrentMonth(addMonths(currentMonth, 1))
    }
    const prevMonth= () => {
        setCurrentMonth(subMonths(currentMonth, 1))
    }


    return (
        <>
            
            <div className='calendar'>
                {renderHeader()}
                {renderDays()}
                {renderCells()}
            </div>
            
        </>
    )
}

export default Calendar