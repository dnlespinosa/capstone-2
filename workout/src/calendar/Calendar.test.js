import React from "react";
import { render } from "@testing-library/react";
import Calendar from "./Calendar";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <Calendar />
        </UserProvider>
      </MemoryRouter>,
  );
});

  // snapshot 
  it('matches snapshot', function() {
    const { asFragment } = render(
            <MemoryRouter>
                <UserProvider>
                    <Calendar />
                </UserProvider>
            </MemoryRouter>
        )
    expect(asFragment()).toMatchSnapshot();
  })