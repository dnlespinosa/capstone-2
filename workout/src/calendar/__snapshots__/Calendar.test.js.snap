// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="calendar"
  >
    <h1>
      YOUR PROGRAM
    </h1>
    <div
      class="header row flex-middle"
    >
      <div
        class="col col-start"
      >
        <div
          class="icon"
        >
          chevron_left
        </div>
      </div>
      <div
        class="col col-center"
      >
        <span>
          June 02023
        </span>
      </div>
      <div
        class="col col-end"
      >
        <div
          class="icon"
        >
          chevron_right
        </div>
      </div>
    </div>
    <div
      class="days row"
    >
      <div
        class="col col-center"
      >
        0025
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="row"
      >
        <div
          class="col cell disabled"
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            28
          </span>
        </div>
        <div
          class="col cell disabled"
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            29
          </span>
        </div>
        <div
          class="col cell disabled"
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            30
          </span>
        </div>
        <div
          class="col cell disabled"
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            31
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            1
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            2
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            3
          </span>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            4
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            5
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            6
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            7
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            8
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            9
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            10
          </span>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            11
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            12
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            13
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            14
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            15
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            16
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            17
          </span>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            18
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            19
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            20
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            21
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            22
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            23
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            24
          </span>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            25
          </span>
        </div>
        <div
          class="col cell selected"
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            26
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            27
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            28
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            29
          </span>
        </div>
        <div
          class="col cell "
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            30
          </span>
        </div>
        <div
          class="col cell disabled"
        >
          <div>
            <a
              href="/personal-workout/undefined"
            >
              <p
                id="displayedWorkout"
              />
            </a>
          </div>
          <span
            class="number"
          >
            1
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
